source/grammar/index.rst

```diff
@@ -1,12 +1,132 @@
 Grammar
 =======
 
-Grammar specification based in ANTLR_ parser generator
+This is a simplified grammar for Open QASM presented in Backus-Naur
+form. The unlisted productions :math:`\langle\mathrm{id}\rangle`,
+:math:`\langle\mathrm{real}\rangle` and
+:math:`\langle\mathrm{nninteger}\rangle` are defined by the regular
+expressions below::
 
-.. _ANTLR: https://www.antlr.org/
+		id        := [a-z][A-Za-z0-9_]*
+		real      := ([0-9]+\.[0-9]*|[0-9]*\.[0-9]+)([eE][-+]?[0-9]+)?
+		nninteger := [1-9]+[0-9]*|0
+        ...       := .*
 
-[WIP]
+Production rules are defined with plain text and denoted by a colon ``production : rule``.
+Keywords are enclosed in quotations ``rule : "keyword"``. Whitespace denotes
+concatenation. Alternatives are denoted by ``|`` and may be grouped with 
+parentheses ``(rule1 | rule2 | ...)``. Optional rules are terminated with a ``?`` (zero or one), 
+repetition is denoted by curly brackets ``+`` (one or more) and ``*`` (zero or more). An ellipses
+``...`` is used to match any series of token.
 
-.. literalinclude:: qasm3.g4
-   :language: antlr
-   :linenos:
\ No newline at end of file
+.. productionlist::
+    program: header generic_statement*
+    header: version? include*
+    version: "OPENQASM" real ";"
+    include: "include" id ( ".inc" | ".qasm" ) ";"
+    generic_statement: global_statement
+        :| statement
+    global_statement: subroutine_declaration
+        :| kernel_declaration
+        :| gate_definition
+        :| calibration
+    statement: expression_statement
+        :| declaration_statement
+        :| selection_statement
+        :| iteration_statement
+        :| selection_directive_statement
+        :| quantum_statement
+        :| timing_box
+        :| pragma
+    subroutine_declaration: "def" id args_declaration return_signature? program_block
+    args_declaration : "(" type_and_id_list ")"
```

Comment:
```suggestion
    subroutine_args : ( "(" type_and_id_list? ")" )?
```

Reply:
why? this is more general and applies to kernel/subroutine